specification/compute/resource-manager/Microsoft.Compute/preview/2018-06-01/gallery.json

```diff
@@ -0,0 +1,1120 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "GalleryClient",
+    "description": "The Gallery Client.",
+    "version": "2018-06-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}": {
+      "put": {
+        "tags": [
+          "Galleries"
+        ],
+        "operationId": "Galleries_CreateOrUpdate",
+        "description": "Create or update a gallery.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gallery."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "gallery",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Gallery"
+            },
+            "description": "Parameters supplied to the create or update gallery operation."
+          }
+        ],
+        "responses": {
+          "201": {
```

Comment:
200 ok response needs to be modeled. What if a PUT on an already created resource happens. In that case, it should return a 200ok

Reply:
In our API spec, a PUT on an already created resource will update the resource, which returns 202 accepted. A PUT on a nonexistent resource will return 201. So I wonder if this still needs to be change. Here is our API spec: https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=%2Fsrc%2FCAPS%2FDoc%2FPrivateGalleryDesignAndApiDoc.md&version=GBdevelop

Reply:
Why does PUT on an existing resource where nothing has changed result in a 202?

Reply:
This is fixed.